freeplane/src/main/java/org/freeplane/view/swing/map/overview/MapOverviewUtils.java

```diff
@@ -0,0 +1,100 @@
+        } catch (IllegalArgumentException e) {
+            return MapOverviewAttachPoint.SOUTH_EAST;
+        }
+    }
+
+    public static void convertOriginByAttachPoint(MapView mapView, Rectangle bounds) {
+        final JScrollPane mapViewScrollPane = (JScrollPane) (mapView.getParent().getParent());
+        Insets insets = mapViewScrollPane.getInsets();
+        int bottom = mapViewScrollPane.getHeight() - insets.bottom;
+        int right = mapViewScrollPane.getWidth() - insets.right;
+        JScrollBar hsb = mapViewScrollPane.getHorizontalScrollBar();
+        JScrollBar vsb = mapViewScrollPane.getVerticalScrollBar();
+        int vsw = vsb.isVisible() ? vsb.getSize().width : 0;
+        int hsw = hsb.isVisible() ? hsb.getSize().height : 0;
+
+        switch (getMapOverviewAttachPoint()) {
+        case SOUTH_EAST:
+            bounds.setLocation(right - bounds.x - vsw - 1 - bounds.width, bottom - bounds.y - hsw - 1 - bounds.height);
+            break;
+        case SOUTH_WEST:
+            bounds.setLocation(bounds.x, bottom - bounds.y - hsw - 1 - bounds.height);
+            break;
+        case NORTH_EAST:
+            bounds.setLocation(right - bounds.x - vsw - 1 - bounds.width, bounds.y);
+            break;
+        case NORTH_WEST:
+            break;
+        }
+    }
+
+    public static int getBoundedValue(int value, int min, int max) {
+        return Math.max(Math.min(value, max), min);
+    }
+
+    public static Rectangle getMapOverviewBounds(MapView mapView) {
+        ResourceController resourceController = ResourceController.getResourceController();
+        String rawBoundsValue = resourceController.getProperty(MapOverviewConstants.PROP_MAP_OVERVIEW_BOUNDS);
+        int[] elements = null;
+        if (rawBoundsValue != null) {
+            String[] rawElements = rawBoundsValue.split(",");
+            if (rawElements.length == 4) {
+                elements = new int[4];
+                for (int i = 0; i < 4; i++) {
+                    try {
+                        elements[i] = Integer.parseInt(rawElements[i]);
+                    } catch (NumberFormatException e) {
+                        elements = null;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (elements == null) {
+            elements = new int[] { 0, 0, MapOverviewConstants.DEFAULT_SIZE, MapOverviewConstants.DEFAULT_SIZE };
+        }
+        int x = elements[0];
+        int y = elements[1];
+        int width = getBoundedValue(elements[2], MapOverviewConstants.MIN_SIZE, MapOverviewConstants.MAX_SIZE);
+        int height = getBoundedValue(elements[3], MapOverviewConstants.MIN_SIZE, MapOverviewConstants.MAX_SIZE);
+        Rectangle bounds = new Rectangle(x, y, width, height);
+        convertOriginByAttachPoint(mapView, bounds);
+        return bounds;
+    }
+
+    public static double getBestScale(Rectangle source, double scaledOverviewBoundWidth,
```

Comment:
I would move this method to MapOverviewImage and remove its first argument.

Reply:
Already moved into MapOverviewImage, but could you reconsider moving the first parameter, I might want to keep it since mapView.getInnerBounds() seems to be none trivial, so I'd calculate only once and pass it down.

I can understand that from clean code perspective too much parameters are bad so if possible I'll try to reduce number of parameters of method.

Reply:
Reducing the number of parameters usually leads either to either moving logic to the data or to creating new abstractions.
I see your argument that you want to avoid expansive calculation of the map view inner bounds.
Probably MapView could cache the results until they are invalidated and remove the cache when MapView. paintChildren(final Graphics g) is called. But it is only something to be considered, as freeplane code is not clean enough to have all good principles satisfied. So I think we can leave it as it is now.